src/core/ext/filters/client_channel/resolver/polling_resolver.cc

```diff
@@ -100,29 +101,22 @@ void PollingResolver::ShutdownLocked() {
   }
   shutdown_ = true;
   if (have_next_resolution_timer_) {
-    grpc_timer_cancel(&next_resolution_timer_);
+    event_engine_->Cancel(next_resolution_timer_handle_);
   }
   request_.reset();
 }
 
-void PollingResolver::OnNextResolution(void* arg, grpc_error_handle error) {
-  auto* self = static_cast<PollingResolver*>(arg);
-  self->work_serializer_->Run(
-      [self, error]() { self->OnNextResolutionLocked(error); }, DEBUG_LOCATION);
-}
-
-void PollingResolver::OnNextResolutionLocked(grpc_error_handle error) {
+void PollingResolver::OnNextResolutionLocked() {
   if (GPR_UNLIKELY(tracer_ != nullptr && tracer_->enabled())) {
     gpr_log(GPR_INFO,
-            "[polling resolver %p] re-resolution timer fired: error=\"%s\", "
-            "shutdown_=%d",
-            this, StatusToString(error).c_str(), shutdown_);
+            "[polling resolver %p] re-resolution timer fired: shutdown_=%d",
+            this, shutdown_);
   }
   have_next_resolution_timer_ = false;
-  if (error.ok() && !shutdown_) {
+  if (!shutdown_) {
     StartResolvingLocked();
   }
-  Unref(DEBUG_LOCATION, "retry-timer");
+  Unref(DEBUG_LOCATION, "retry_timer");
```

Comment:
This `Unref` can be removed if the timer callback owns a Ref local variable, the unreffing will be handled automatically on closure destruction. See the RunAfter usage in `src/core/ext/filters/client_channel/lb_policy/weighted_target/weighted_target.cc`.

Reply:
Done... I'm such a newbie...